### PR TITLE
Python 3.8 compatibility

### PR DESCRIPTION
**Pull request by jannisborn**
This PR enables compatibility with Python 3.8 for GT4SD.
Let's merge it after #151 so that the diff shrinks.

- I forked moler and relaxed its dependencies, it works out of the box for 3.8 🚀 
- mypy works slightly differently in 3.8 so had to fix some stuff
- Had to update `torchmetrics` bounds since otherwise a conflict with lightning occurs. Basically the same problem as in #146, but now caused by lightning requiring a newer version of `torchmetrics`. It only occurs in editable mode installation that's why #149 passed the CI.
- I did not verify functionality for Python 3.9, so to play save I set an upper bound
- I still observe issues on MacOS when running torchdrug inference  in a fresh installation (training is ok), potentially it's a local problem (it seems related to `torch_extensions`)::
```
E   ImportError: dlopen(/Users/jab/Library/Caches/torch_extensions/py38_cpu/torch_ext_0/torch_ext.so, 0x0002): 
tried: '/Users/jab/Library/Caches/torch_extensions/py38_cpu/torch_ext_0/torch_ext.so' (no such file)
```

### CODE REVIEW

File: src/gt4sd/training_pipelines/__init__.py [hunk 1]
```diff
@@ -208,16 +210,15 @@ def training_pipeline_name_to_metadata(name: str) -> Dict[str, Any]:
     Returns:
         dictionary describing the parameters of the pipeline. If the pipeline is not found, no metadata (a.k.a., an empty dictionary is returned).
     """
+    file_manager = ExitStack()
```
Comment (drugilsberg) [inline src/gt4sd/training_pipelines/__init__.py:213]: see comments for PR #151

File: src/gt4sd/training_pipelines/tests/test_training_moses_vae.py [hunk 2]
```diff
@@ -23,19 +23,21 @@
...
-import pkg_resources
+import importlib_resources
 
 from gt4sd.training_pipelines import TRAINING_PIPELINE_MAPPING, MosesVAETrainingPipeline
 
-TEST_DATA_DIRECTORY = pkg_resources.resource_filename(
-    "gt4sd",
-    "training_pipelines/tests/",
-)
+file_manager = ExitStack()
```
Comment (drugilsberg) [inline src/gt4sd/training_pipelines/tests/test_training_moses_vae.py:37]: see comments for PR #151

File: vcs_requirements.txt [hunk 3]
```diff
@@ -1,6 +1,6 @@
 # version control systems requirements
 guacamol_baselines @ git+https://github.com/GT4SD/guacamol_baselines.git@v0.0.2
-molecule_generation @ git+https://github.com/microsoft/molecule-generation.git@v0.2.0
+molecule_generation @ git+https://github.com/GT4SD/molecule-generation.git@main
```
Comment (drugilsberg) [inline vcs_requirements.txt:3]: cool, if there were no major changes, we could even open a PR on the original repo to suggest the update

File: conda.yml [hunk 4]
```diff
@@ -5,7 +5,7 @@ channels:
   - https://conda.anaconda.org/pyg
   - https://conda.anaconda.org/conda-forge
 dependencies:
-  - python>=3.7,<3.8
+  - python>=3.7,<3.9
```
Comment (drugilsberg) [inline conda.yml:8]: good idea, we can consider testing with 3.9 for a later PR and eventually relax the bound

**Review by drugilsberg — COMMENTED** [review]
Great work, pending changes on #151

File: setup.cfg [hunk 5]
```diff
@@ -11,13 +11,16 @@ classifiers =
...
+    Programming Language :: Python :: 3.8
 
 [options]
 install_requires =
     accelerate
     datasets
     diffusers
     ipaddress
+    importlib_metadata
+    importlib_resources
```
Comment (georgosgeorgos) [inline setup.cfg:23]:
```suggestion
    importlib_metadata
    importlib_resources
    ipaddress
```

File: requirements.txt [hunk 6]
```diff
@@ -4,7 +4,9 @@ datasets>=1.11.0
 diffusers>=0.2.4
 ipaddress>=1.0.23
 joblib>=1.1.0
```
Comment (georgosgeorgos) [inline requirements.txt:6]:
```suggestion
importlib-metadata>=1.7.0,<5.0.0 # temporary: https://github.com/python/importlib_metadata/issues/409
importlib-resources>=5.10.0
ipaddress>=1.0.23
joblib>=1.1.0
```

File: requirements.txt [hunk 7]
```diff
@@ -4,7 +4,9 @@ datasets>=1.11.0
...
-keras==2.3.1
+importlib-metadata>=1.7.0,<5.0.0 # temporary: https://github.com/python/importlib_metadata/issues/409
+importlib-resources>=5.10.0
```
Comment (georgosgeorgos) [inline requirements.txt:8]:
```suggestion
```

**Review by georgosgeorgos — APPROVED** [review]
Great work! Just a comment on the imports.

File: vcs_requirements.txt (hunk 3 above)
Comment (jannisborn) [reply vcs_requirements.txt:3]: Doing that now!

**Review by drugilsberg — APPROVED** [review]
Looks great. Let's wait for the CI and then we can move on with a release.

**Pull request merged**